content/blogposts/2022/sourcegraph-4.0.md

```diff
@@ -0,0 +1,247 @@
+
+<Video 
+  source={{
+    webm: 'blog/release-post/4.0/Reference%20panel',
+    mp4: 'blog/release-post/4.0/Reference%20panel'
+  }}
+  loop={true}
+  title="Reference panel refresh"
+  caption="Stay within your workflow and quickly access references with the revamped reference panel"
+/>
+
+<a href="https://docs.sourcegraph.com/code_navigation/explanations/features#find-references" className="tw-not-italic tw-flex tw-items-center">Docs<OpenInNewIcon className="tw-ml-xxs" size={18} /></a>
+
+
+
+### 🏗️ High-leverage ways to improve your entire codebase
+<br />
+<Badge link="/batch-changes" text="Batch Changes" color="blue" size="small" />
+
+#### Make changes across your codebase at enterprise scale with server-side Batch Changes (beta)
+[Batch Changes](https://about.sourcegraph.com/batch-changes) allows you to automate code changes across your entire codebase. Previously, you had to run `src-cli` locally to create batch changes, which could take an impractical amount of time or be brittle for large or resource-intensive batch changes. Now, with server-side Batch Changes, large-scale changes can be run across thousands of repositories with a better development experience. With server-side Batch Changes, you can:
+
+- Run large-scale or resource-intensive batch changes without clogging your local machine.
+- Run large batch changes quickly by distributing them across an autoscaled pool of compute instances.
+- Get a better debugging experience by streaming logs directly into Sourcegraph.
+
+This feature requires admins to set up executors (much like CI agents), which Sourcegraph uses for running large-scale changes. Executors can also be used to run [code navigation auto-indexing](https://github.com/sourcegraph/about/blob/bf341aafde75450c295aaec064470017020cdcd8/content/blogposts/2022/sourcegraph-4.0.md#TODO-link-to-section-or-docs). All Sourcegraph Cloud instances have one or more executors available by default. 
+
+<Video 
+  source={{
+    webm: 'blog/release-post/4.0/Server%20Side%20Batch%20Changes%20',
+    mp4: 'blog/release-post/4.0/Server%20Side%20Batch%20Changes%20'
+  }}
+  loop={true}
+  title="Server-side Batch Changes"
+  caption="Large-scale changes can be run across thousands of repositories with server-side Batch Changes"
+/>
+
+<a href="https://docs.sourcegraph.com/batch_changes/explanations/server_side" className="tw-not-italic tw-flex tw-items-center">Docs<OpenInNewIcon className="tw-ml-xxs" size={18} /></a>
+
+### ☁️ Dedicated Sourcegraph Cloud instances for enterprise
+
+<br />
+<Badge link="https://docs.sourcegraph.com/cloud" text="Admin" color="violet" size="small" />
+
+####  Sourcegraph Cloud, our single-tenant cloud offering, is now generally available
```

Comment:
@iskyOS one final review, please.

Reply:
Done!

Reply:
@iskyOS @lcolston08 Now that the cloud post has been updated, thoughts on this (to make it clear that we technically already have a cloud offering:

After almost a decade of developing self-hosted Sourcegraph, we’ve gained the trust of some of the most sophisticated companies and development teams in the world, allowing us to work with their most sensitive IP: their code. But not everyone loves self-hosting, so over the last year we worked with customers to build a Sourcegraph cloud service that meets their security and scalability needs and we recently completed our [SOC 2 Type II audit](https://security.sourcegraph.com).

Dedicated, single-tenant Sourcegraph Cloud instances are now generally available and the best way for teams to use Sourcegraph. You can read more in our announcement blog [here](/blog/enterprise-cloud), or [sign up for a 30-day trial for your organization](https://signup.sourcegraph.com/).